packages/office-ui-fabric-react/src/components/SpinButton/SpinButton.tsx

```diff
@@ -31,16 +31,18 @@ export interface ISpinButtonState {
    * as active when up/down arrow is pressed
    */
   keyboardSpinDirection: KeyboardSpinDirection;
-
-  /**
-   * The calculated precision for the value.
-   */
-  precision: number;
 }
 
+export type DefaultProps = Required<
+  Pick<ISpinButtonProps, 'step' | 'min' | 'max' | 'disabled' | 'labelPosition' | 'label' | 'incrementButtonIcon' | 'decrementButtonIcon'>
+>;
+
+/** Internal only props */
+type ISpinButtonInternalProps = ISpinButtonProps & DefaultProps;
+
 @customizable('SpinButton', ['theme', 'styles'], true)
 export class SpinButton extends BaseComponent<ISpinButtonProps, ISpinButtonState> implements ISpinButton {
-  public static defaultProps: ISpinButtonProps = {
+  public static defaultProps: DefaultProps = {
```

Comment:
don't understand this. What is this change for?

Reply:
Had to create a separate DefaultProps that makes the optional props required within the class itself. All of this helped me get rid of the if/else undefined checks & the constant this.props usage.